cli/Sources/TuistKit/Services/LoginService.swift

```diff
@@ -79,12 +90,35 @@ final class LoginService: LoginServicing {
                 password: password,
                 serverURL: serverURL
             )
+        } else if Environment.current.isCI {
+            try await authenticateWithCIOIDC(serverURL: serverURL, onEvent: onEvent)
         } else {
             try await authenticateWithBrowserLogin(serverURL: serverURL, onEvent: onEvent)
         }
         await onEvent(.completed)
     }
 
+    private func authenticateWithCIOIDC(
+        serverURL: URL,
+        onEvent: @escaping (LoginServiceEvent) async -> Void
+    ) async throws {
+        await onEvent(.oidcAuthenticating)
+
+        let oidcToken = try await ciOIDCAuthenticator.fetchOIDCToken()
+        let accessToken = try await exchangeOIDCTokenService.exchangeOIDCToken(
+            oidcToken: oidcToken,
+            serverURL: serverURL
+        )
+
+        try await ServerCredentialsStore.current.store(
+            credentials: ServerCredentials(
+                accessToken: accessToken,
+                refreshToken: ""
+            ),
```

Comment:
The `refreshToken: ""` reads a bit like trying to reuse a symbol, `ServerCredentials`, with a new use case it wasn't designed for, persisted account tokens generated from OIDC tokens. I'd consider evolving the type.